internal/imageprovider/imageprovider.go

```diff
@@ -1,95 +1,120 @@
-// imageprovider.go: contains code for fetching bird images from an image provider
+// imageprovider.go: Package imageprovider provides functionality for fetching and caching bird images.
 package imageprovider
 
 import (
+	"log"
 	"sync"
+	"time"
 	"unsafe"
 
 	"github.com/tphakala/birdnet-go/internal/telemetry"
+	"github.com/tphakala/birdnet-go/internal/telemetry/metrics"
 )
 
+// ImageProvider defines the interface for fetching bird images.
 type ImageProvider interface {
-	fetch(scientificName string) (BirdImage, error)
+	Fetch(scientificName string) (BirdImage, error)
```

Comment:
Why is this function made public?

Reply:
For unit tests to work

Reply:
Can't the unit tests be in the same package as before?